### PR TITLE
Add Jakarta EE 8 support

### PR DESCRIPTION
**Pull request by hantsy**
see https://github.com/graphql-java/graphql-java-extended-validation/pull/53#issuecomment-957122700 and https://github.com/graphql-java/graphql-java-extended-validation/pull/53#issuecomment-957146265

### CODE REVIEW

**Comment by hantsy** [comment]
fixes #52

File: jakarta.gradle [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+task copyTestJavaSources(type: Copy) {
+    into generatedTestSrc
+    from 'src/test/java'
+    filter { line -> line.replaceAll("javax\\.", "jakarta\\.") }
+}
+
+task copyJavaSources(type: Copy) {
+    into generatedSrc
+    from 'src/main/java'
+    filter { line -> line.replaceAll("javax\\.", "jakarta\\.") }
```
Comment (pelletier197) [inline jakarta.gradle:25]: That's a bit a funny hack to make it work. I don't mind much personally as this is rather temporary, but are we sure there are no changes between `javax` and `jakarta` outside the package name ?

**Review by pelletier197 — COMMENTED** [review]
Looks good, and will fix issues for many people, including me. Two questions for your

- Should we update the readme to outline the existence of this classifier?
- I'm not familiar with how this library is deployed, but will they need to do something special to deploy it ?

File: build.gradle [hunk 2]
```diff
@@ -60,6 +71,11 @@ task javadocJar(type: Jar, dependsOn: javadoc) {
     from javadoc.destinationDir
 }
 
+jar {
+    // add jakarta classier for Jakarta EE 9
+    archiveClassifier = isJakarta ? "jakarta" : ""
```
Comment (pelletier197) [inline build.gradle:76]: is emtpy string the default value, or should it be null?

File: jakarta.gradle (hunk 1 above)
Comment (hantsy) [reply jakarta.gradle:25]: I am sure for Java EE 8/Jakarta EE 9 developers there is no difference from the API level between Jakarta EE 8 and Jakarta EE 9, the biggest change is the namespace migrating to new **jakarta**.

File: build.gradle (hunk 2 above)
Comment (hantsy) [reply build.gradle:76]: I am not sure the default value of the *classifier*, I think empty string works.

**Comment by hantsy** [comment]
@pelletier197 Thanks for reviewing my PRs. 

**But I need more help to polish this PR to make it more mature,  as the check list in the original post.**

Personally I hope to build a special **jakarta** classifier version for Jakarta EE 9 , and keep the default version for Jakarta EE 8, but it seems the MavenPublishPlugin can set it dynamically. 

 And if possible to add a new archive(**jakarta**) for Jakarta EE 9 besides the default archives(src, docs, jar, and a new jar with **jakarta** classifier).

**Comment by sbilello** [comment]
Is there any update on this PR? So we can unblock: https://github.com/Netflix/dgs-framework/pull/659

**Comment by hantsy** [comment]
Another possible solution is maintaining a branch for Jakarta EE 8,and make the main/master to Jakarta EE 9. This is maybe the most common options to maintain multi versions.

**Comment by hantsy** [comment]
@pelletier197 @sbilello Tried to create two branches(jakartaee8,  jakartaee9) on my fork, the `build.gradle` looks more clear.

* The `jakartaee8` branch will need to sync to upstream `jakartaee8`, and maintain a `jakartaee8` branch in future.
* The `jakartaee9` branch just add postfix **jakarta** the artifact name to identify the a new namespace.

Looks more clear?

Hope the maintainer of this project can work together and decide the final solution.

**Comment by sbilello** [comment]
If we can make a change that would make it clear and keep `jakartaee8` working: it should allow any project to avoid dealing with runtime dependency conflicts. I guess it is a good workaround  to avoid to upgrade `jakartaee8` in the Spring Framework

**Comment by pelletier197** [comment]
I'm personally completely okay with this and I would merge as is.

Unfortunately I'm not a maintainer of this repo, so even if we can merge it, I cannot deploy it. And the maintainers of the repo are not really reactive to PRs. I have not had any update on my recent PRs as well for a while.

> I was personally able to use the library with Jakarta in spring. Addong Jakarta to your classpath should resolve this, but it is indeed a dirty workaround.

**Comment by bbakerman** [comment]
> Another possible solution is maintaining a branch for Jakarta EE 8,and make the main/master for Jakarta EE 9. This is maybe the most common options to maintain multi versions.

I think this is a much better option that gradle config that edits files at runtime.  I think a branch is a much better option

**Comment by hantsy** [comment]
> > Another possible solution is maintaining a branch for Jakarta EE 8,and make the main/master for Jakarta EE 9. This is maybe the most common options to maintain multi versions.
> 
> I think this is a much better option that gradle config that edits files at runtime. I think a branch is a much better option

This is ready in my side, there is a **jakartaee8** branch required to be created on this project,, thus I can create new PRs for different branches.

**Comment by bbakerman** [comment]
I ended up making a PR and branch and released a new version


https://github.com/graphql-java/graphql-java-extended-validation/pull/54

https://github.com/graphql-java/graphql-java-extended-validation/releases/tag/17.0-hibernate-validator-6.2.0.Final


So we should be right to go.  Please test that release (its in Maven now) and tell me if there are any problems

**Pull request closed without merging**